### PR TITLE
Introduce TransactionSimulatorService

### PR DESCRIPTION
**Pull request by fab-10**
Introduce TransactionSimulatorService so plugins can simulate transactions on their own

### CODE REVIEW

File: besu/src/main/java/org/hyperledger/besu/services/TransactionSimulationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    final CallParameter callParameter = CallParameter.fromTransaction(transaction);
+
+    final var blockHeader =
+        blockchain
+            .getBlockHeader(blockHash)
+            .or(() -> blockchain.getBlockHeaderSafe(blockHash))
+            .orElseThrow(
+                () ->
+                    new IllegalStateException(
+                        "Block header not yet present for chaih head hash: " + blockHash));
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/services/TransactionSimulationServiceImpl.java:63]:
```suggestion
                        "Block header not yet present for chain head hash: " + blockHash));
```

File: plugin-api/src/main/java/org/hyperledger/besu/plugin/data/TransactionSimulationResult.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright ConsenSys AG.
```
Comment (macfarla) [inline plugin-api/src/main/java/org/hyperledger/besu/plugin/data/TransactionSimulationResult.java:2]: should this be the besu copyright

Comment (fab-10) [reply plugin-api/src/main/java/org/hyperledger/besu/plugin/data/TransactionSimulationResult.java]: fixed

File: besu/src/main/java/org/hyperledger/besu/services/TransactionSimulationServiceImpl.java (hunk 1 above)
Comment (fab-10) [reply besu/src/main/java/org/hyperledger/besu/services/TransactionSimulationServiceImpl.java]: done

**Comment by fab-10** [comment]
Thanks @macfarla for the review, I merged this to unlock the Linea release in testnet, but feel free to submit other comments if needed, that I will address later.

**Pull request merged**